### PR TITLE
fix: Harmonize handling of JS

### LINKED ISSUE #9374: Harmonize handling of JS
### What
- Harmonize handling of JS
```
<script src="https://static.openfoodfacts.org/js/dist/foundation.js" data-base-layout="true"></script> 
<script src="https://static.openfoodfacts.org/js/dist/jquery.cookie.js"></script> 
<script src="https://static.openfoodfacts.org/js/dist/select2.min.js"></script> 
<script src="/js/datatables.min.js"></script> 
<script src="/js/dist/papaparse.js"></script>
```

### PR DESCRIPTION
**Pull request by mike-lu1**
### What
I have replaced the occurrence of `<script src="https://static.openfoodfacts.org/js` and <script src="/js with  `<script src="[% static_subdomain %]/js`

### Screenshot

[Screencast from 12-03-24 02:32:02.webm](https://github.com/openfoodfacts/openfoodfacts-server/assets/162586393/101b03d9-f50b-437a-826f-16aece88a43f)

### Related issue(s) and discussion

- Fixes #9374

### CODE REVIEW

**Comment by mike-lu1** [comment]
hey @stephanegigandet @alexgarel kindly review this.

File: cgi/import_photos_upload.pl [hunk 1]
```diff
@@ -155,24 +155,24 @@
     </script>
 
     <!-- The Templates plugin is included to render the upload/download listings -->
-    <script src="/js/dist/tmpl.js"></script>
+    <script src="[% static_subdomain %]/js/dist/tmpl.js"></script>
```
Comment (stephanegigandet) [inline cgi/import_photos_upload.pl:158]: [% static_subdomain %] only works for templates that are processed by Template::Toolkit (files in /templates/).

In Perl code, you need to use $static_subdomain instead

**Review by stephanegigandet — CHANGES_REQUESTED** [review]
If you try it locally, there are JS errors in the console because the files can't be loaded:

crostata-artigianale-forno-miotti:2862          GET http://fr.openfoodfacts.localhost/produit/8033162870633/[%%20static_subdomain%20%]/js/product-search.js net::ERR_ABORTED 400 (Bad Request)

in the source generated:

<script src="[[% static_subdomain %]/js/product-preferences.js](http://fr.openfoodfacts.localhost/produit/8033162870633/[%%20static_subdomain%20%]/js/product-preferences.js)"></script>
<script src="[[% static_subdomain %]/js/product-search.js](http://fr.openfoodfacts.localhost/produit/8033162870633/[%%20static_subdomain%20%]/js/product-search.js)"></script>

Could you try to fix it and then test it locally to verify there are no JS errors?

File: html/donate/aa.html [hunk 2]
```diff
@@ -87,7 +87,7 @@
         z-index: 0;
       }
     </style>
-    <script src="https://static.openfoodfacts.org/js/dist/jquery.js"></script>
+    <script src="[% static_subdomain %]/js/dist/jquery.js"></script>
```
Comment (hangy) [inline html/donate/aa.html:90]: The static pages in `html/donate` cannot use the `Template::Toolkit` variables. They probably need to stay as-is

**Comment by mike-lu1** [comment]
> Could you try to fix it and then test it locally to verify there are no JS errors?

@stephanegigandet the `products-history.js` is having some persistent jQuery error `Uncaught ReferenceError: $ is not defined`
I have have tweaked the function in most of the ways i know but this has actually not worked for me 

At the parent function ie fist reference of the `$` in the .js file, i don't get the error but i appears in child functions

**Comment by stephanegigandet** [comment]
> @stephanegigandet the `products-history.js` is having some persistent jQuery error `Uncaught ReferenceError: $ is not defined`

This is a bug unrelated to your changes. It was fixed in this PR: https://github.com/openfoodfacts/openfoodfacts-server/pull/9909

If you update your branch with the main branch, it should solve this error.

**Comment by mike-lu1** [comment]
thank you @hangy @stephanegigandet.  I have implemented the above suggestions

File: cgi/product_multilingual.pl [hunk 3]
```diff
@@ -789,7 +789,7 @@ ($product_ref, $field, $language)
 var admin = $moderator;
 </script>
 <script type="text/javascript" src="$static_subdomain/js/dist/product-multilingual.js?v=$file_timestamps{'js/dist/product-multilingual.js'}"></script>
-<script type="text/javascript" src="$static_subdomain/js/dist/product-history.js"></script>
+<script type="text/javascript" src="/js/dist/product-history.js"></script>
```
Comment (stephanegigandet) [inline cgi/product_multilingual.pl:792]: I don't think this one should be removed.

**Review by stephanegigandet — APPROVED** [review]
Thank you!

**Pull request merged**